### PR TITLE
Add markdown documentation

### PR DESCRIPTION
**Pull request by psyyz10**
## What changes were proposed in this pull request?

Add markdown documentation

### CODE REVIEW

File: docs/docs/APIdocs/layers/nn/Loss Functions/ClassNLLCriterion.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+## ClassNLLCriterion
+
+**Scala:**
+```scala
+val criterion = ClassNLLCriterion()
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Loss Functions/ClassNLLCriterion.md]: perhaps  ClassNLLCriterion[T]()

File: docs/docs/APIdocs/layers/nn/Loss Functions/ClassNLLCriterion.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+              T(2f, 3f, 4f),
+              T(3f, 4f, 5f)
+          ))
+
+val target = Tensor(T(1f, 2f, 3f))
+
+val loss = criterion.forward(input, target)
+val grad = criterion.backward(input, target)
+
+print(nloss)
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Loss Functions/ClassNLLCriterion.md]: not nloss, but loss?

File: docs/docs/APIdocs/layers/nn/Merge Layers/JoinTable.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
+## JoinTable
+**Scala:**
+```scala
+val layer = JoinTable(2, 2)
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Merge Layers/JoinTable.md]: should be JoinTable[T](dimension, nInputDims)

File: docs/docs/APIdocs/layers/nn/Loss Functions/ClassNLLCriterion.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+The negative log likelihood criterion. It is useful to train a classification problem with n
+classes. If provided, the optional argument weights should be a 1D Tensor assigning weight to
+each of the classes. This is particularly useful when you have an unbalanced training set.
+
+The input given through a forward() is expected to contain log-probabilities of each class:
+input has to be a 1D Tensor of size n. Obtaining log-probabilities in a neural network is easily
+achieved by adding a LogSoftMax layer in the last layer of your neural network. You may use
+CrossEntropyCriterion instead, if you prefer not to add an extra layer to your network. This
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Loss Functions/ClassNLLCriterion.md]: Shall we quote (\` \`) functions, parameters, class names, etc.? e.g. `forward`, `n`, and `backward` and `LogSoftMax`, `CrossEntropyCriterion`

File: docs/docs/APIdocs/layers/nn/Loss Functions/ClassNLLCriterion.md (hunk 1 above)
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Loss Functions/ClassNLLCriterion.md]: What is the default value if we don't specify [T]? Double?

File: docs/docs/APIdocs/layers/nn/Merge Layers/JoinTable.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+It is a table module which takes a table of Tensors as input and
+outputs a Tensor by joining them together along the dimension `dimension`.
+
+
+The input to this layer is expected to be a tensor, or a batch of tensors;
+when using mini-batch, a batch of sample tensors will be passed to the layer and
+the user need to specify the number of dimensions of each sample tensor in the
+batch using `nInputDims`.
+
+**Parameters:**
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Merge Layers/JoinTable.md]: do we need to specify parameters?

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/HardTanh.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
+## HardTanh
+
+**Scala:**
+```scala
+val activation = HardTanh()
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/HardTanh.md]: I think we should specify paramters in constructor.

File: docs/docs/APIdocs/layers/nn/Loss Functions/ClassNLLCriterion.md (hunk 1 above)
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Loss Functions/ClassNLLCriterion.md]: should we specify all parameters in constructor and default values if any?

**Comment by zhangxiaoli73** [comment]
LGTM now
